### PR TITLE
Initial project

### PR DESCRIPTION
**Pull request by alebabai**
Core bring up

### CODE REVIEW

**Review by MickevichYura — DISMISSED** [review]

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        <plugins>
+            <plugin>
+                <groupId>org.springframework.boot</groupId>
+                <artifactId>spring-boot-maven-plugin</artifactId>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>11</source>
```
Comment (solairerove) [inline pom.xml]: you've already inited `<java.version>11</java.version>` in properties. use it here pls as `${java.version}`

Comment (solairerove) [inline pom.xml]: maybe it would be better to use last version of maven compiler plugin and last `<compilerVersion>` tag

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    <description>Core Service</description>
+
+    <properties>
+        <java.version>11</java.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-webflux</artifactId>
```
Comment (solairerove) [inline pom.xml:24]: i've just thought that we will use standard web implementation

File: src/main/java/com/github/meeteor13/core/Application.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.github.meeteor13.core;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class Application {
```
Comment (solairerove) [inline src/main/java/com/github/meeteor13/core/Application.java:7]: Rename to `CoreApplication` or whatever (`LegacyApplication`) ))

File: src/main/java/com/github/meeteor13/core/Application.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+    public static void main(String[] args) {
```
Comment (solairerove) [inline src/main/java/com/github/meeteor13/core/Application.java]: what about `final` keyword to args plz.
By the way, maybe you'll provide us local [sonarqube container](https://github.com/solairerove/docker/tree/master/sonarqube)

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+.idea/caches/build_file_checksums.ser
+
+# JetBrains templates
+**___jb_tmp___
+
+### Intellij+all Patch ###
+# Ignores the whole .idea folder and all .iml files
+# See https://github.com/joeblau/gitignore.io/issues/186 and https://github.com/joeblau/gitignore.io/issues/360
+
+.idea/
```
Comment (MickevichYura) [inline .gitignore]: если есть это, то зачем всё остальное с .idea/**?

**Review by MickevichYura — APPROVED** [review]

**Review by MickevichYura — APPROVED** [review]

**Review by MickevichYura — DISMISSED** [review]

**Review by solairerove — CHANGES_REQUESTED** [review]
5 реквест ченжей висит. деклайн или откоменьтся

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+# [ Production image ]
+# ===============================
+FROM openjdk:11-jre-slim
+
+LABEL name="meeteor13-core" \
+      maintainer="aliaksandr.babai@gmail.com"
+
+ARG WORK_DIR=/meeteor13
+
+ENV SERVER_PORT=80 \
```
Comment (vadim-gleyf) [inline Dockerfile:23]: why we should define port for app in dockerfile?

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    JAVA_OPTS=""
```
Comment (vadim-gleyf) [inline Dockerfile:24]: remove it

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,15 @@
+
+version: "3.7"
+services:
+  core:
+    container_name: core
+    image: meeteor13/core:latest
+    build: .
+    ports:
+      - 80:8080
```
Comment (vadim-gleyf) [inline docker-compose.yml]: did you test it?
HOST:CONTAINER
in container you define 80 port

File: Dockerfile [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+WORKDIR $WORK_DIR
+COPY --from=BUILDER $WORK_DIR/target/*.jar .
+
+EXPOSE $SERVER_PORT
+ENTRYPOINT ["/bin/sh", "-c", "java $JAVA_OPTS -Djava.security.egd=file:/dev/urandom -jar *.jar"]
```
Comment (vadim-gleyf) [inline Dockerfile]: why you add all this shit instead just
`ENTRYPOINT java -jar *.jar`

File: docker-compose.yml (hunk 8 above)
Comment (vadim-gleyf) [reply docker-compose.yml]: ENV > app.yml

File: pom.xml [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+        <relativePath/> <!-- lookup parent from repository -->
+    </parent>
+    <groupId>com.github.meeteor13</groupId>
+    <artifactId>core</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
+    <name>Core</name>
+    <description>Core Service</description>
+
+    <properties>
+        <java.version>11</java.version>
```
Comment (vadim-gleyf) [inline pom.xml:18]: why not 12?

File: pom.xml (hunk 1 above)
Comment (vadim-gleyf) [reply pom.xml]: agree, pls remove duplicates

File: src/main/resources/application.yml [hunk 11]
```diff
@@ -0,0 +1,2 @@
+server:
+  port: 8080
```
Comment (vadim-gleyf) [inline src/main/resources/application.yml:2]: it's default value, remove it

File: .editorconfig [hunk 12]
```diff
@@ -0,0 +1,10 @@
+root = true
```
Comment (vadim-gleyf) [inline .editorconfig:1]: maybe we should add some checks on compile?
https://github.com/ec4j/editorconfig-maven-plugin
something like this, by default idea doesn't look on this file, i should install plugin

**Review by vadim-gleyf — CHANGES_REQUESTED** [review]
why we don't use jdk 12?

File: src/main/java/com/github/meeteor13/core/Application.java (hunk 3 above)
Comment (alebabai) [reply src/main/java/com/github/meeteor13/core/Application.java:7]: I'm not agree with u.
Classes shouldn't contains name of the product (service).

File: Dockerfile (hunk 9 above)
Comment (alebabai) [reply Dockerfile]: stupid grebec
please read matchast'

Comment (alebabai) [reply Dockerfile]: it will not work.
this shit `*.jar` is processed by shell

File: pom.xml (hunk 10 above)
Comment (alebabai) [reply pom.xml:18]: there isn't good docker image for that.

File: src/main/resources/application.yml (hunk 11 above)
Comment (alebabai) [reply src/main/resources/application.yml:2]: don't agree, it should be there to be self documented (to know which port will be used)

File: .editorconfig (hunk 12 above)
Comment (alebabai) [reply .editorconfig:1]: it has only 5 starts - not enterprise levev

File: src/main/java/com/github/meeteor13/core/Application.java (hunk 3 above)
Comment (vadim-gleyf) [reply src/main/java/com/github/meeteor13/core/Application.java:7]: @alebabai but you tests contains

File: pom.xml (hunk 2 above)
Comment (alebabai) [reply pom.xml:24]: No, we decided to start with newest stuff

File: .editorconfig (hunk 12 above)
Comment (vadim-gleyf) [reply .editorconfig:1]: it's just example
there are no any checks, i can write code in any way
we should add checks or remove this file

File: src/main/resources/application.yml (hunk 11 above)
Comment (vadim-gleyf) [reply src/main/resources/application.yml:2]: so, why you override only port with default value, what about all other props?

Comment (alebabai) [reply src/main/resources/application.yml:2]: Port in application.yml used for development.
Port in docker container with web service should be 80, it's defacto standard.

File: .editorconfig (hunk 12 above)
Comment (alebabai) [reply .editorconfig:1]: editorconfig used for development, to have common config for our IDE (it's called EDITORconfig).
It's not a linter.

File: Dockerfile (hunk 7 above)
Comment (alebabai) [reply Dockerfile:24]: read about _OPTS in any scripts

File: src/main/java/com/github/meeteor13/core/Application.java (hunk 3 above)
Comment (alebabai) [reply src/main/java/com/github/meeteor13/core/Application.java:7]: yeap, good point, i'll fix

File: .editorconfig (hunk 12 above)
Comment (vadim-gleyf) [reply .editorconfig:1]: by default ide doesn't use it, isn't it?
i should install a plugin (and i can not install)

Comment (alebabai) [reply .editorconfig:1]: Yep u should install plugin, it's just for convenience

Comment (alebabai) [reply .editorconfig:1]: it's not a hard rule and not a linter

File: pom.xml (hunk 1 above)
Comment (alebabai) [reply pom.xml]: As i know it handled by spring boot parent.

**Comment by alebabai** [comment]
are u happy?

**Review by MickevichYura — APPROVED** [review]

**Review by solairerove — APPROVED** [review]

**Review by vadim-gleyf — APPROVED** [review]

**Pull request merged**